src/Storages/ObjectStorageQueue/ObjectStorageQueueMetadata.cpp

```diff
@@ -924,6 +925,13 @@ void ObjectStorageQueueMetadata::unregisterNonActive(const StorageID & storage_i
             return;
         }
 
+        if (!responses.empty() && supports_remove_recursive && code == Coordination::Error::ZNOTEMPTY) /// potentially we reached RemoveRecursive node limit, let's try without it
```

Comment:
Could we have partially succeeded? Like some of the nodes were actually removed? If we did remove  `registry_path`, but not other paths, then it will quit immediately without retrying the removal,

Reply:
RemoveRecursive is atomic